library_information_validator.py

```diff
@@ -0,0 +1,495 @@
+        else:
+            sindex_chosen = False
+
+            # warning if a pool consists out of a mix of NGI standard indexes and custom indexes
+            sel_index = sindex_list + cindex_list
+            if(len(sel_index) != len(pool_rows)):
+                rowID_cindex_absent = [re.sub("\D","",x) for x in cindex_absent]
+                rowID_sindex_absent = [re.sub("\D","",x) for x in sindex_absent]
+
+                missing_indexes = list(set(rowID_cindex_absent).intersection(rowID_sindex_absent))
+                for mIndex in missing_indexes:
+                    logger.error('missing index in row {}'.format(mIndex))
+
+            logger.warning('mix between custom and standard indexes in pool {}.'\
+            .format(pool))
+            warning_mixed_indexes += 1
+
+        # returns warnings for check summary
+        return(sel_index, sindex_chosen, warning_mixed_indexes, warning_component_length)
+
+    def validate_index(self, index_seq, pool_name, sindex):
+        '''
+        does all the fancy index checks
+        '''
+
+        # allows for entry "noIndex" if only one sample is defined in the pool
+        c = Counter(index_seq)
+        if(c['noIndex'] > 0 and len(index_seq) != 1):
+            logger.error('Pool {} contains undefined index(es) (\"noIndex\")'\
+            .format(pool_name))
+            quit()
+        elif(c['noIndex'] == 1 and len(index_seq) == 1):
+            logger.info('Pool {} containing one sample is not indexed.'\
+            .format(pool_name))
+        elif(c['noIndex'] == 0):
+            # checks that all indexes in a pool are unique
+            for index, index_count in c.most_common():
+                if(index_count>1):
+                    logger.error('The index sequence \"{}\" in pool {} is not unique for this pool.'\
+                    .format(index, pool_name))
+                    quit()
+                else:
+                    break
+
+        warning_low_div = 0
+        index_count = 1
+        for index in index_seq:
+            # checks that indexes only contain valid letters
+            charRE = re.compile(r'[^ATCGNatcgn\-.]')
+            index_search = charRE.search(index)
+            if(bool(index_search) and index != "noIndex"):
+                logger.error('The index sequence \"{}\" in pool {} contains invalid characters.'\
+                ' Allowed characters: A/T/C/G/N/a/t/c/g/n/-'
+                .format(index, pool_name))
+                quit()
+
+            # check that indexes within a pool have minimum diversity
+            for i in range(index_count,len(index_seq)):
+                if lev.distance(index.lower(), index_seq[i].lower()) < LibrarySheet.MAX_DISTANCE:
+                    logger.warning('The index sequences {} and {} in pool {}'\
+                    ' display low diversity (only {} nt difference).'\
+                    .format(index,index_seq[i], pool_name, lev.distance(index.lower(), index_seq[i].lower())))
```

Comment:
You calculate the same Lev. distance twice if the test returns true, if it's an expensive operation maybe you should store the result from `lev.distance` in a variable for re-use.